corehq/motech/repeaters/optionvalue.py

```diff
@@ -106,6 +109,9 @@ def __set__(self, obj, value):
             value = self.coder.to_json(value)
         obj.options[self.name] = value
 
+    def get_default_value(self):
+        return self.default() if callable(self.default) else self.default
```

Comment:
I forgot part of this in my previous suggestion, I think it should raise if the default is `NOT_SET`
```suggestion
        if self.default is self.NOT_SET:
            raise AttributeError(self.name)
        return self.default() if callable(self.default) else self.default
```
And lines 92-93 can be removed above. To your reason:

> the reason I returned `None` was the returned JSON should at least have all the keys that are there in the Model.

I agree, but this will only ever be called on valid models, right? If the model is valid then all required fields should have values. If it is not valid, we don't want `NOT_SET` values in the JSON. That seems like it would be an error.

Reply:
> I agree, but this will only ever be called on valid models, right? If the model is valid then all required fields should have values. If it is not valid, we don't want NOT_SET values in the JSON. That seems like it would be an error.

Yep yep. It makes sense. I have updated the code accordingly.